app/recipeApp.js

```diff
@@ -1,26 +1,55 @@
-let recipeObject
-
-let recipes = []; 
+let recipes =[];
 
 async function getData() {
-    const response = await fetch(
-        "https://raw.githubusercontent.com/jianxinz233/jianxinz233.github.io/refs/heads/main/data/recipe_app_data.json"
-    );
-    recipes  = await response.json();
-    showAllRecipes();
-}
-
-getData();
+    try{
+        const response = await fetch(
+            "https://raw.githubusercontent.com/jianxinz233/jianxinz233.github.io/refs/heads/main/data/recipe_app_data.json"
+        ); 
+        const recipeData  = await response.json();
+        recipes = recipeData;
+        return recipes;
+    } catch (error) {
+        console.error(`We encountered an error: ${error}`);
```

Comment:
do you want to throw an error here? 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw